cmd/soroban-cli/src/config/network.rs

```diff
@@ -106,6 +136,24 @@ pub struct Network {
     pub network_passphrase: String,
 }
 
+fn parse_http_header(header: &str) -> Result<(String, String), Error> {
+    let header_components = header.split(':').collect::<Vec<&str>>();
+    if header_components.len() != 2 {
+        return Err(Error::InvalidHeader(format!(
+            "Missing a header name and/or value: {header}"
+        )));
+    }
+
+    let key = header_components[0].trim().to_string();
+    let value = header_components[1].trim().to_string();
```

Comment:
I think technically `:` can appear in the header value, so splitting and exploding that into a Vec on every `:` is probably not a resilient, as splitting on the first `:` and otherwise keeping the rest of the header intact.

It should be a small change to something like:

```suggestion
    let header_components = header.split(':');
    let Some(key) = header_components.next() else {
        return Err(Error::InvalidHeader(format!(
            "Missing a header name and/or value: {header}"
        )));
    };
    let value = header_components.remaining();
```

Reply:
That makes sense to me - the one piece that I'm noticing is that `remainder` is only available with the unstable nightly build at the moment. Though, I think we can get a similar effect by just using `next` again. 
https://doc.rust-lang.org/std/str/struct.Split.html#impl-Split%3C'a,+P%3E

Reply:
Ah perfect, that's great 👏🏻.